src/copy.c

```diff
@@ -239,9 +257,23 @@ TSCopyMultiInsertBufferFlush(TSCopyMultiInsertInfo *miinfo, TSCopyMultiInsertBuf
 	CommandId mycid = miinfo->mycid;
 	int ti_options = miinfo->ti_options;
 	int nused = buffer->nused;
-	ResultRelInfo *resultRelInfo = buffer->cis->result_relation_info;
 	TupleTableSlot **slots = buffer->slots;
 
+	/*
```

Comment:
I feel confused now -- what is the lifetime of TSCopyMultiInsertBuffer, who owns it, and how it compares to the lifetime of ChunkInsertState? I think we should have a comment  somewhere explaining this.

The following code looks like lifetime of insert buffer is not bound to lifetime of chunk insert state, but who owns it then?

Reply:
I would imagine that it is owned by chunk insert state and flushed when chunk insert state is discarded or when the insert ends. Not sure if it's workable in practice.

Reply:
Ah, OK, it's owned by `TSCopyMultiInsertBufferFlush` and flushed when it's full. So indeed it looks right to just look up the chunk insert state again when flushing.

Reply:
Its a good point. I made this more clear and cleared the memory context of the outdated CIS accordingly.